duffy/models/openstack_nodes.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+from duffy.database import db, Duffyv1Model
+from duffy.extensions import marshmallow
+from nodes import *
+
+
+class openstack_host(Duffyv1Model):
+    __tablename__ = 'openstack_nodes'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(30))
+    hostname = db.Column(db.String(20))
+    ip = db.Column(db.String(15))
+    state = db.Column(db.String(20))
+    next_state = db.Column(db.String(20))
+    comment = db.Column(db.String(255))
+    image_source = db.Column(db.String(30))
+    flavor = db.Column(db.Sting(20))
+    session_id = db.Column(db.String(37), db.ForeignKey('sessions.id'))
+    session = db.relationship('Session', lazy='joined')
+
+    def contextualize(self, project):
+        self.state = 'Contextualizing'
+        self.save()
+        # Sync all of the keys to the root user on the remote host
+        import paramiko
+        import os
+        try:
+            ssh = paramiko.SSHClient()
+            # TODO: Make this configurable
+            key = paramiko.DSSKey.from_private_key_file(os.path.expanduser('~/.ssh/id_dsa'))
+            ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+            ssh.connect(self.ip, username='root', pkey=key)
```

Comment:
We should check to be sure which username we need to use for the openstack user account. This may be a wider issue actually, since we won't know about this beforehand. Example: for CentOS images, the root account is disabled, and we need to use the 'centos' user account

Reply:
will adding cloud-init while provisioning help here?
I am not sure how it will take usernames
is there any template described in openstack config?